libass/ass_fontselect.c

```diff
@@ -166,7 +168,91 @@ static ASS_FontProviderFuncs ft_funcs = {
     .destroy_font      = destroy_font_ft,
 };
 
-static void load_fonts_from_dir(ASS_Library *library, const char *dir)
+
+#if defined(_WIN32) && !defined(__CYGWIN__)
+static void load_fonts_from_dir(ASS_Library* library, const char* dir)
+{
+    WIN32_FIND_DATAW find_file_context;
+    HANDLE find_file_handle;
+    UINT cur_cp = CP_UTF8;
+
+    char* valid_search_dir;
+    int tmp;
+
+    if (!dir || !(tmp = strlen(dir)))
+        return;
+
+    valid_search_dir = (char*)malloc(sizeof(char) * (tmp + 2));
+    strcpy(valid_search_dir, dir);
+    switch (valid_search_dir[tmp - 1]) {
+    case '\\':
+    case '/':
+    case ':':
+        break;
+    default:
+        valid_search_dir[tmp++] = '\\';
+    }
+    valid_search_dir[tmp++] = '*';
+    valid_search_dir[tmp] = '\0';
+
+    wchar_t* dir_utf16le = ass_convert_to_utf16le(valid_search_dir, cur_cp);
+
+    if (!dir_utf16le || (find_file_handle = FindFirstFileW(dir_utf16le, &find_file_context)) == INVALID_HANDLE_VALUE) {
+        free(dir_utf16le);
+        cur_cp = CP_ACP;
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+        if (!AreFileApisANSI()) cur_cp = CP_OEMCP;
+#endif
+        dir_utf16le = ass_convert_to_utf16le(valid_search_dir, cur_cp);
+        if (dir_utf16le) find_file_handle = FindFirstFileW(dir_utf16le, &find_file_context);
```

Comment:
The handle opened by `FindFirstFileW` never gets [closed](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findclose).